### PR TITLE
create db from snapshots

### PR DESCRIPTION
**Pull request by ChrisScottThomas**
This changes RDS creation to use snapshots to recreate itself.

- RDS now produces its own final snapshot on removal.
- RDS encrypts storage.
- The infra creates, and RDS uses a KMS key to read/write.
- Simplified the cluster name, to make this easily repeatable.
- Bumped the PostgreSQL version to `10.11`

### CODE REVIEW

File: terraform/modules/database/rds.tf [hunk 1]
```diff
@@ -2,8 +2,23 @@ resource "aws_db_subnet_group" "cluster" {
   subnet_ids = var.vpc.aws_subnets_private[*].id
 }
 
+resource "aws_kms_key" "aurora" {
+  enable_key_rotation = true
+  tags                = merge(var.tags, { Name = "${var.name}-db-key" })
```
Comment (jimbarker) [inline terraform/modules/database/rds.tf:7]: Could do with a ProtectsSensitiveData tag too, as per the Tagging policy.

File: terraform/modules/database/rds.tf [hunk 2]
```diff
@@ -2,8 +2,23 @@ resource "aws_db_subnet_group" "cluster" {
   subnet_ids = var.vpc.aws_subnets_private[*].id
 }
 
+resource "aws_kms_key" "aurora" {
```
Comment (jimbarker) [inline terraform/modules/database/rds.tf:5]: deletion_window_in_days = 7 will help to quiet some of the noise from AWS Config if this key needs to be deleted.

File: terraform/modules/database/rds.tf (hunk 1 above)
Comment (ChrisScottThomas) [reply terraform/modules/database/rds.tf]: Done.

File: terraform/modules/database/rds.tf (hunk 2 above)
Comment (ChrisScottThomas) [reply terraform/modules/database/rds.tf:5]: Done.

File: terraform/modules/database/rds.tf [hunk 3]
```diff
@@ -2,8 +2,24 @@ resource "aws_db_subnet_group" "cluster" {
...
+  tags                    = merge(var.tags, { Name = "${var.name}-db-key", ProtectsSensitiveData = true })
+}
+
+resource "aws_kms_alias" "aurora" {
+  name          = "alias/${var.name}-db-key"
+  target_key_id = aws_kms_key.aurora.key_id
+}
+
+data "aws_db_cluster_snapshot" "cluster" {
+  db_cluster_snapshot_identifier = var.name
```
Comment (dan-hill2802) [inline terraform/modules/database/rds.tf:17]: might be a be better coupled to the resource
```suggestion
  db_cluster_snapshot_identifier = aws_rds_cluster.cluster.cluster_identifier
```

**Review by jimbarker — DISMISSED** [review]

File: terraform/modules/database/rds.tf (hunk 3 above)
Comment (dan-hill2802) [reply terraform/modules/database/rds.tf]: this would be a cyclical dependencies, so ignore me

**Review by mattburgess — APPROVED** [review]

**Review by mattburgess — APPROVED** [review]

**Pull request merged**